### PR TITLE
Added option to get all users from LDAP

### PR DESCRIPTION
**Pull request by diospark**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/openmrs/openmrs-contrib-id/pull/88" target="_blank">ReviewNinja

**Comment by robbyoconnor** [comment]
Did you run the tests?  Go look at the travis logs ;) your code broke things.

**Comment by robbyoconnor** [comment]
Also, Write tests for this.

**Comment by diospark** [comment]
Sorry, no

I see the problem, and let you know when I'll done

**Comment by robbyoconnor** [comment]
In order for this to be merged, your code will need to be covered by tests and be passing.

**Comment by robbyoconnor** [comment]
Travis build is fixed, where are the new tests to cover the new functionality you added?

File: tests/ldap.js [hunk 1]
```diff
@@ -369,5 +369,33 @@ describe('ldap', function() {
...
   });
+  
+  describe('#usersList', function() {
+    before(function (done) {
+      return done();
+    });
+    after(function (done) {
+      ldap.deleteUser(VALID_USER.username, done);
+    });
+    it('should return a list with one more user than before', function (done) {
```
Comment (robbyoconnor) [inline tests/ldap.js]: No, this should return all users. In addition to this, your test is still failing 
!fixed

File: tests/ldap.js [hunk 2]
```diff
@@ -369,5 +369,33 @@ describe('ldap', function() {
...
+      ldap.getAllUsers(function(err, prevUsers) {
+        expect(err).to.not.exist;
+        var prevCount = prevUsers ? prevUsers.length : 0 ;
```
Comment (robbyoconnor) [inline tests/ldap.js]: This should be the total numbr of users in the system. The tests need to be self-contained. !fixed

**Comment by dkayiwa** [comment]
@diospark are you still working on this?

**Comment by robbyoconnor** [comment]
@dkayiwa, please do not merge code into ID Dashboard. I am manually merging this.

**Comment by robbyoconnor** [comment]
@diospark, could you squash these commits into one?

**Comment by robbyoconnor** [comment]
Strike that-- I can do it :)

**Comment by robbyoconnor** [comment]
I did it -- I'm manually merging this by hand (well cherry-picking -- there's something really elegant about a linear commit history :)

**Comment by robbyoconnor** [comment]
I cherry-picked and squashed your PR and the resulting commit is  11d1d2ac60c0d84087e61a9dd7a3085e1c488f8f.  You're going to need to do a hard reset (`git reset --hard upstream/master`)

**Pull request closed without merging**